pkg/controllers/hub/membercluster/controller.go

```diff
@@ -59,15 +60,22 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		klog.ErrorS(err, "Failed to get memberCluster", "memberCluster", mcObjRef)
 		return ctrl.Result{}, err
 	}
+	if mc.DeletionTimestamp == nil {
```

Comment:
```suggestion
	if mc.DeletionTimestamp.IsZero() {
```
it's better use "mc.DeletionTimestamp.IsZero()" in case it is set but zero.